api/app/db/crud/weather_models.py

```diff
@@ -224,6 +224,26 @@ def get_historic_station_model_predictions(
     return query
 
 
+def get_station_model_prediction_from_previous_model_run(
+        session: Session,
+        station_codes: List[int],
+        model: str,
+        prediction_timestamp: str,
+        prediction_model_run_timestamp: str):
+    """ Fetches the one model prediction for the specified station_code, model, and prediction_timestamp
+    from the prediction model run immediately previous to the given prediction_model_run_timestamp.
+    """
+    query = session.query(WeatherStationModelPrediction, PredictionModelRunTimestamp, PredictionModel).\
+        filter(WeatherStationModelPrediction.station_code.in_(station_codes)).\
+        filter(WeatherStationModelPrediction.prediction_timestamp == prediction_timestamp).\
+        filter(PredictionModelRunTimestamp.prediction_model_id == PredictionModel.id, PredictionModel.abbreviation == model).\
+        filter(PredictionModelRunTimestamp.prediction_run_timestamp < prediction_model_run_timestamp).\
+        order_by(PredictionModelRunTimestamp.prediction_run_timestamp.desc()).\
+        first()
+
+    return query
```

Comment:
Technically this isn't a query at this point, since you've called first, it's the response.

Reply:
Renamed variable